src/main/java/com/scalar/db/storage/jdbc/JdbcDatabase.java

```diff
@@ -0,0 +1,188 @@
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.storage.NoMutationException;
+import com.scalar.db.storage.jdbc.checker.OperationChecker;
+import com.scalar.db.storage.jdbc.metadata.TableMetadataManager;
+import com.scalar.db.storage.jdbc.query.QueryBuilder;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A storage implementation with JDBC for {@link DistributedStorage}.
+ *
+ * <p>Note that the consistency in an operation is always LINEARIZABLE in this implementation. Even
+ * if consistency is specified in an operation, it will be ignored.
+ *
+ * @author Toshihiro Suzuki
+ */
+@ThreadSafe
+public class JdbcDatabase implements DistributedStorage {
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcDatabase.class);
+
+  private final BasicDataSource dataSource;
+  private final JdbcService jdbcService;
+  private Optional<String> namespace;
+  private Optional<String> tableName;
+
+  @Inject
+  public JdbcDatabase(DatabaseConfig config) {
+    dataSource = JdbcUtils.initDataSource(config);
+    Optional<String> namespacePrefix = config.getNamespacePrefix();
+    TableMetadataManager tableMetadataManager =
+        new TableMetadataManager(dataSource, namespacePrefix);
+    OperationChecker operationChecker = new OperationChecker(tableMetadataManager);
+    QueryBuilder queryBuilder =
+        new QueryBuilder(
+            tableMetadataManager, JdbcUtils.getRdbEngine(config.getContactPoints().get(0)));
+    jdbcService = new JdbcService(operationChecker, queryBuilder, namespacePrefix);
+    namespace = Optional.empty();
+    tableName = Optional.empty();
+  }
+
+  @VisibleForTesting
+  JdbcDatabase(BasicDataSource dataSource, JdbcService jdbcService) {
+    this.dataSource = dataSource;
+    this.jdbcService = jdbcService;
+  }
+
+  @Override
+  public void with(String namespace, String tableName) {
+    this.namespace = Optional.ofNullable(namespace);
+    this.tableName = Optional.ofNullable(tableName);
+  }
+
+  @Override
+  public void withNamespace(String namespace) {
+    this.namespace = Optional.ofNullable(namespace);
+  }
+
+  @Override
+  public Optional<String> getNamespace() {
+    return namespace;
+  }
+
+  @Override
+  public void withTable(String tableName) {
+    this.tableName = Optional.ofNullable(tableName);
+  }
+
+  @Override
+  public Optional<String> getTable() {
+    return tableName;
+  }
+
+  @Override
+  public Optional<Result> get(Get get) throws ExecutionException {
+    try (Connection connection = dataSource.getConnection()) {
+      return jdbcService.get(get, connection, namespace, tableName);
+    } catch (SQLException e) {
+      throw new ExecutionException("An error occurred", e);
```

Comment:
I think the message should be a bit more specific because `An error occurred` is obvious and not very informational.
I left a similar comment on the JdbcTransaction code, so please take a look!